MatrixSDK/Crypto/MXCrypto.m

```diff
@@ -1051,16 +1053,11 @@ - (void)flushNewDeviceRequests
         // tight-loop.
         for (NSString *userId in usersDevicesInfoMap.userIds)
         {
-            [pendingNewDevices removeObjectsForUser:userId];
+            [pendingUsersWithNewDevices removeObject:userId];
```

Comment:
I think this is subject to a race:

* we start a device list download
* the other user starts a new device, and we get an m.new_device event for the new device
* we get the response to the device list download, but it doesn't yet include the new device
* we clear `pendingUsersWithNewDevices`, but we still don't know about the new device

The js-sdk clears the pending list before starting the device list download to avoid this.